tenable/io/v3/mssp/logos/api.py

```diff
@@ -0,0 +1,229 @@
+.. autoclass:: LogosAPI
+    :members:
+'''
+from io import BytesIO
+from typing import BinaryIO, Dict, List, Optional, Union
+from uuid import UUID
+
+from tenable.io.v3.base.endpoints.explore import ExploreBaseEndpoint
+from tenable.io.v3.mssp.logos.schema import LogoSchema
+
+
+class LogosAPI(ExploreBaseEndpoint):
+    '''
+    This class contains methods related to Logos API
+    '''
+    _path = 'api/v3/mssp/logos'
+    _conv_json = True
+    _schema = LogoSchema()
+
+    def add(self, logo: Union[BytesIO, BinaryIO], name: str) -> UUID:
+        '''
+        Adds a new logo
+
+        Args:
+            fobj (FileObject, optional):
+                A file-like object of the logo.
+            name (str):
+                Name of the logo.
+
+        Returns:
+            :obj:`str`:
+                Return the logo_id.
+
+        Example:
+            >>>> with open('/opt/download1.png', 'rb') as fobj:
+            ....    resp = tio.v3.mssp.logos.add(fobj, 'logo1')
+        '''
+        payload = {'files': {'logo': logo, 'name': name}}
+        return self._post(**payload)['id']
+
+    def list(self):
+        '''
+        Returns list of logos
+
+        Args:
+
+        Returns:
+            :obj:`list`:
+                Returns all the logo details as a list.
+
+        Example:
+            >>>> tio.v3.mssp.logos.list()
+        '''
+        return self._get()['logos']
+
+    def search(self):
+        '''
+        Get the listing of Logos from MSSP portal.
+
+        Args:
+            *filters (tuple, optional):
+                Filters are tuples in the form of
+                ('NAME', 'OPERATOR', 'VALUE'). Multiple filters can be used
+                to filter down the data being returned from the API.
+                Examples:
+                    - ``('name', 'eq', 'example')``
+            limit (int, optional):
+                The number of records to retrieve.  Default is 50
+            sort (tuple, optional):
+                A tuple of tuples identifying the the field and sort order of
+                the field.
+            fields (list):
+                list of fields to retrieve.
+        Returns:
+            :obj:`LogosIterator`:
+                An iterator that handles the page management of the requested
+                records.
+        Examples:
+            Todo
+        '''
+        raise NotImplementedError('Search method will be implemented later')
+
+    def details(self, logo_id: UUID) -> Dict:
+        '''
+        Returns Logo details
+
+        Args:
+            logo_id (UUID):
+                The UUID of the logo
+        Returns:
+            :obj:`dict`:
+                Return the logo details.
+
+        Example:
+            >>>> tio.v3.mssp.logos.details(
+            ....    '61a36add-d29b-4a52-bbce-c8215952ede5')
+        '''
+        return self._get(f'{logo_id}')
+
+    def update(self, logo_id: UUID, **kw) -> str:
+        '''
+        Returns Logo details
```

Comment:
Updates a logo in the Tenable.io MSSP Portal. This update overwrites the existing logo.